optuna_dashboard/static/components/GraphContour.tsx

```diff
@@ -0,0 +1,259 @@
+                  </MenuItem>
+                ))}
+              </Select>
+            </FormControl>
+          ) : null}
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter1">X Axis Parameter</InputLabel>
+            <Select value={xAxis} onChange={handleXAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+          <FormControl component="fieldset" className={classes.formControl}>
+            <InputLabel id="parameter2">Y Axis Parameter</InputLabel>
+            <Select value={yAxis} onChange={handleYAxisChange}>
+              {paramnames.map((x) => (
+                <MenuItem value={x} key={x}>
+                  {x}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid>
+      </Grid>
+      <Grid item xs={6}>
+        <div id={plotDomId} />
+      </Grid>
+    </Grid>
+  )
+}
+
+const plotContour = (
+  study: StudyDetail,
+  objectiveId: number,
+  xAxis: string,
+  yAxis: string
+) => {
+  if (document.getElementById(plotDomId) === null) {
+    return
+  }
+
+  const layout: Partial<plotly.Layout> = {
+    title: "Contour",
+    margin: {
+      l: 50,
+      r: 50,
+    },
+    xaxis: {
+      gridcolor: "#f2f5fa",
+      gridwidth: 1,
+      zerolinecolor: "#f2f5fa",
+      zerolinewidth: 1.5,
+    },
+    yaxis: {
+      gridcolor: "#f2f5fa",
+      gridwidth: 1,
+      zerolinecolor: "#f2f5fa",
+      zerolinewidth: 1.5,
+    },
+    plot_bgcolor: "#E5ecf6",
+  }
+
+  const trials: Trial[] = study !== null ? study.trials : []
+
+  if (trials.length === 0) {
+    plotly.react(plotDomId, [], layout)
+  }
+
+  const filteredTrials = trials.filter(
+    (t) => t.state === "Complete" || t.state === "Pruned"
+  )
+
+  let paramNames = new Set<string>(trials[0].params.map((p) => p.name))
+  filteredTrials.forEach((t) => {
+    paramNames = new Set<string>(
+      t.params.filter((p) => paramNames.has(p.name)).map((p) => p.name)
+    )
+  })
+
+  if (paramNames.size === 0 || paramNames.size === 1) {
+    plotly.react(plotDomId, [], layout)
+    return
+  }
+  if (xAxis === yAxis) {
+    plotly.react(plotDomId, [], layout)
+    return
+  }
+
+  const objectiveValues: number[] = filteredTrials.map(
+    (t) => t.values![objectiveId]
+  )
+
+  const paramValues: { [key: number]: string[] } = []
+  const paramIndices: Array<string> = []
+
+  if (paramNames.size >= 2) {
+    let i = 0
+    paramNames.forEach((paramName) => {
+      const valueStrings = filteredTrials.map((t) => {
+        const param = t.params.find((p) => p.name == paramName)
+        return param!.value
+      })
+      paramValues[i] = valueStrings
+      i++
+    })
+
+    let k = 0
+    paramNames.forEach((paramName) => {
+      paramIndices[k] = paramName
+      k++
+    })
```

Comment:
You can get the index number from the second argument of the callback function.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

Reply:
Sounds good! I'll make the required change.